### PR TITLE
INF-367 Asset hash deploys for pxt-microbit

### PR DESCRIPTION
**Pull request by wryun**
[INF-367]

Add redirect so we can still hit up the main website

[INF-367]: https://groklearning.atlassian.net/browse/INF-367

### CODE REVIEW

File: buildspec.yaml [hunk 1]
```diff
@@ -11,18 +11,17 @@ phases:
...
       # https://makecode.com/cli/staticpkg
-      - node node_modules/pxt-core/built/pxt.js staticpkg -m
+      - node node_modules/pxt-core/built/pxt.js staticpkg -m -r "$DEPLOY_HASH"
       # It might be cute to put the following in post_build, but unfortunately you have
       # to check for a successful build in that case... (CODEBUILD_BUILD_SUCCEEDING)
       - BRANCH="$(git branch -a --contains HEAD | grep -vF -e '(no branch)' -e 'origin' | head -1 | sed 's/^..//')"
       - echo On branch "$BRANCH"...
       - if [ "$BRANCH" = develop -o "$BRANCH" = master ]; then deploy=true; else deploy=false; fi
-      - if $deploy; then aws s3 sync built/packaged "s3://$S3_BUCKET_NAME"; fi
+      - if $deploy; then aws s3 sync built/packaged/$DEPLOY_HASH "s3://$S3_BUCKET_NAME/$DEPLOY_HASH"; fi
```
Comment (timdawborn) [inline buildspec.yaml:26]: Where does `S3_BUCKET_NAME` come from?

File: buildspec.yaml [hunk 2]
```diff
@@ -11,18 +11,17 @@ phases:
...
+      - if $deploy; then aws s3api put-object --bucket pxtmicrobit.dev.comp.gl --key index.html --website-redirect-location "/$DEPLOY_HASH/index.html?controller=1"; fi
```
Comment (timdawborn) [inline buildspec.yaml]: Does this need to be done for prod as well?

Comment (wryun) [reply buildspec.yaml]: err, that should say S3_BUCKET_NAME...

File: buildspec.yaml (hunk 1 above)
Comment (wryun) [reply buildspec.yaml:26]: codebuild - see codebuild/template.yaml in cloudformation repo

**Review by timdawborn — APPROVED** [review]
LGTM :+1:

**Pull request merged**